src/cryptography/hazmat/primitives/ciphers/aead.py

```diff
@@ -52,3 +52,67 @@ def _check_params(self, nonce, data, associated_data):
         utils._check_bytes("associated_data", associated_data)
         if len(nonce) != 12:
             raise ValueError("Nonce must be 12 bytes")
+
+
+class AESCCM(object):
+    def __init__(self, key):
+        utils._check_bytes("key", key)
+        if len(key) not in (16, 24, 32):
+            raise ValueError("AESCCM key must be 128, 192, or 256 bits.")
+
+        self._key = key
+
+        if not backend.aead_cipher_supported(self.__class__):
+            raise exceptions.UnsupportedAlgorithm(
+                "AESCCM is not supported by this version of OpenSSL",
+                exceptions._Reasons.UNSUPPORTED_CIPHER
+            )
+
+    @classmethod
+    def generate_key(cls, bit_length):
+        if not isinstance(bit_length, int):
+            raise TypeError("bit_length must be an integer")
+
+        if bit_length not in (128, 192, 256):
+            raise ValueError("bit_length must be 128, 192, or 256")
+
+        return os.urandom(bit_length // 8)
+
+    def encrypt(self, nonce, data, associated_data, tag_length=16):
+        if associated_data is None:
+            associated_data = b""
+
+        self._check_params(nonce, data, associated_data, tag_length)
+        self._validate_l(nonce, len(data))
+        return backend.aesccm_encrypt(
+            self._key, nonce, data, associated_data, tag_length
+        )
+
+    def decrypt(self, nonce, data, associated_data, tag_length=16):
+        if associated_data is None:
+            associated_data = b""
+
+        self._check_params(nonce, data, associated_data, tag_length)
+        return backend.aesccm_decrypt(
+            self._key, nonce, data, associated_data, tag_length
+        )
+
+    def _validate_l(self, nonce, data_len):
```

Comment:
In the RFC this is named `L` and the description is "Number of octets in length field". It represents the encoding of the length of the encrypted payload into bytes. `_validate_nonce_payload_length` maybe? I feel like `validate_l` is actually a good name given the fact that any debugging is going to necessitate a visit to the RFC.